### PR TITLE
Add feature flag for future use to gate AnnData ingest (SCP-4773)

### PR DESCRIPTION
**Pull request by ehanna4**
Add a feature flag defaulted to false that we can use for gating ingesting AnnData files.

This is necessary to support https://broadworkbench.atlassian.net/browse/SCP-4706 

Testing:
There is not real functionality added with this subtask but you can run the migration script to see the new flag.
- Pull this branch
- Open up terminal and run `bin/rails db:migrate`
- Start a server and open up localhost
- Navigate to the Feature Flags list page: https://localhost:3000/single_cell/feature_flags and see the new feature flag

Screen Shot 2022-11-10 at 3 45 30 PM

### CODE REVIEW

**Review by bistline — APPROVED** [review]

**Review by eweitz — APPROVED** [review]
Code looks good!  Nicely contained.

File: db/migrate/20221110110837_add_feature_flag_for_ingesting_anndata.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class AddFeatureFlagForIngestingAnndata < Mongoid::Migration
+    # mirror of FeatureFlag.rb, so this migration won't error if that class is renamed/altered
+    class FeatureFlagMigrator
```
[the remark this answers is not in the document]
Comment (eweitz) [reply db/migrate/20221110110837_add_feature_flag_for_ingesting_anndata.rb]: (The same code caused me to double-take for exactly the same reason, FWIW.  Ideally it'd be removed from all feature flag modules, but keeping Jon's point in institutional memory seems safer and easier.)

File: db/migrate/20221110110837_add_feature_flag_for_ingesting_anndata.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      include Mongoid::Document
+      store_in collection: 'feature_flags'
+      field :name, type: String
+      field :default_value, type: Boolean, default: false
+      field :description, type: String
+    end
+  
+    def self.up
+      FeatureFlagMigrator.create!(name: 'add_feature_flag_for_ingesting_anndata',
```
[the remark this answers is not in the document]
Comment (ehanna4) [reply db/migrate/20221110110837_add_feature_flag_for_ingesting_anndata.rb]: Oh that's a great question - I think I lean slightly towards keeping just one flag. Having fewer flags to remember to toggle feels like it'd help with fewer permutations to keep in our heads for testing and remembering to remove. I think using only one could also help with regression testing, as we build up adding each datatype ingestion on top the next we'd be continuously testing what we'd already built. Also if we start with one now we could always add more later.

[the remark this answers is not in the document]
Comment (ehanna4) [reply db/migrate/20221110110837_add_feature_flag_for_ingesting_anndata.rb]: Oh also yeah that name `add_feature_flag_for_ingesting_anndata` was not good lol, I updated it to `ingest_anndata_file`

[the remark this answers is not in the document]
Comment (bistline) [reply db/migrate/20221110110837_add_feature_flag_for_ingesting_anndata.rb]: > Oh that's a great question - I think I lean slightly towards keeping just one flag. Having fewer flags to remember to toggle feels like it'd help with fewer permutations to keep in our heads for testing and remembering to remove. I think using only one could also help with regression testing, as we build up adding each datatype ingestion on top the next we'd be continuously testing what we'd already built. Also if we start with one now we could always add more later.

Makes sense - good reasoning!

**Review by bistline — APPROVED** [review]

**Pull request merged**